s3-source-connector/src/test/java/io/aiven/kafka/connect/s3/source/input/ParquetTransformerTest.java

```diff
@@ -80,12 +88,15 @@ void testGetRecordsWithValidData() throws Exception {
     void testGetRecordsWithInvalidData() {
         final byte[] invalidData = "invalid data".getBytes(StandardCharsets.UTF_8);
         final InputStream inputStream = new ByteArrayInputStream(invalidData);
+        final IOSupplier<InputStream> inputStreamIOSupplier = () -> inputStream;
+
         final S3SourceConfig s3SourceConfig = mock(S3SourceConfig.class);
 
         final String topic = "test-topic";
         final int topicPartition = 0;
 
-        final List<Object> records = parquetTransformer.getRecords(inputStream, topic, topicPartition, s3SourceConfig);
+        final Stream<Object> records = parquetTransformer.getRecords(inputStreamIOSupplier, topic, topicPartition,
+                s3SourceConfig);
         assertThat(records).isEmpty();
     }
 
```

Comment:
One more test to ensure that the error thrown by getParquetStramRecords always returns Stream.empty() would also be great.